Support for load-env-vars package

diasjorge opened the pull request:
### Brief summary of what the package does

It lets you load a file with environment variable declarations into your current environment inside emacs

### Direct link to the package repository

https://github.com/diasjorge/emacs-load-env-vars

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [X] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks. Quick feedback:

- `filePath` should be `file-path`
- The prompt `"fEnvironment variables file:"` should end with a space
- You should autoload `load-env-vars-from-file` and `load-env-vars-set-env-vars`
- Consider calling `load-env-vars-from-file` simply `load-env-vars`
- The regexp is too simplistic, and doesn't allow for escaped quote characters. Indeed, such variables will then be silently ignored, which will be hard to debug. :-)
- If you make your functions operate on buffers rather than files internally, they'll be easier to debug.

Generally I'd suggest users who want this functionality should prefer the [direnv](https://github.com/wbolster/emacs-direnv) package.

diasjorge commented:
Hi, thanks for the code review. I added your suggestions in this branch here: https://github.com/diasjorge/emacs-load-env-vars/tree/improvements

I'll improve the regexp and add a few more cases.
I also added some tests for the package.

One thing I didn't understand is what do you mean by this? 

> If you make your functions operate on buffers rather than files internally, they'll be easier to debug.

purcell commented:
> One thing I didn't understand is what do you mean by this?
> 
>   >  If you make your functions operate on buffers rather than files internally, they'll be easier to debug.

If you make `load-env-vars-extract-env-vars` operate on a buffer instead of a string, then you can easily test it interactively in an example buffer, and use `M-x re-builder` to experiment with your regexp. You can still wrap it to operate on a file from `load-env-vars-from-file`. That's all. :-)

diasjorge commented:
Hi, so I've made many improvements to the package, thanks for all your suggestions.

Working with buffers simplified a lot of the code, and I ported a much better regexp from the dotenv gem.

One difference regading direnv is that it will read the .envrc file from the current directory. In my case I have many env files for different environments and load them when I need them.

Do you think this is ready to be merged?

purcell commented:
Yep, thanks - merged now. Welcome to MELPA! :tada:
